src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -32,37 +75,231 @@ const NotebookLauncher = _.flow(
+    const { metadata: { lastLockedBy, lockExpiresAt } } = await Ajax(signal).Buckets.notebook(namespace, bucketName, notebookName.slice(0, -6)).getObject()
+    const hashedUser = await notebookLockHash(bucketName, email)
+    const lockExpirationDate = new Date(parseInt(lockExpiresAt))
+
+    if (lastLockedBy && (lastLockedBy !== hashedUser) && (lockExpirationDate > Date.now())) {
+      setLocked(true)
+      setLockedBy(lastLockedBy)
+    }
+  })
+
+  Utils.useOnMount(() => { checkIfLocked() })
+
+  const chooseMode = mode => {
+    Nav.history.replace({ search: '?' + qs.stringify({ mode }) })
+  }
+
+  return div({ style: { display: 'flex', alignItems: 'center', borderBottom: `2px solid ${colors.dark(0.2)}`, height: '3.5rem' } }, [
+    div({ style: { fontSize: 18, fontWeight: 'bold', backgroundColor: colors.dark(0.2), paddingRight: '4rem', paddingLeft: '4rem', height: '100%', display: 'flex', alignItems: 'center' } },
+      ['PREVIEW (READ-ONLY)']),
+    !readOnlyAccess && Utils.cond(
+      [
+        !mode, () => h(Fragment, [
+          h(ButtonSecondary, {
+            style: { paddingRight: '1rem', paddingLeft: '1rem', paddingTop: '1rem', paddingBottom: '1rem', backgroundColor: colors.dark(0.1), height: '100%', marginRight: '2px' },
+            onClick: () => locked ? setFileInUseOpen(true) : chooseMode('edit')
+          }, [icon('lock', { style: { paddingRight: '3px' } }), locked ? 'EDIT (IN USE)' : 'EDIT']),
+          h(ButtonSecondary, {
+            style: { paddingRight: '1rem', paddingLeft: '1rem', backgroundColor: colors.dark(0.1), height: '100%', marginRight: '2px' },
+            onClick: () => {
+              hidePlaygroundModal === 'true' ? chooseMode('playground') : setPlaygroundModalOpen(true)
+            }
+          }, [icon('chalkboard', { style: { paddingRight: '3px' } }), 'PLAYGROUND MODE']),
+          h(ButtonSecondary, {
+            style: { paddingRight: '1rem', paddingLeft: '1rem', backgroundColor: colors.dark(0.1), height: '100%' }
+          }, [
+            h(PopupTrigger, {
+              closeOnClick: true,
+              content: h(Fragment, [
+                h(MenuButton, { onClick: () => setCopyingNotebook(true) }, ['Make a Copy']),
+                h(MenuButton, { onClick: () => setExportingNotebook(true) }, ['Copy to another workspace']),
+                h(MenuButton, {
+                  onClick: async () => {
+                    try {
+                      await clipboard.writeText(`${window.location.host}/${notebookLink}`)
+                      notify('success', 'Successfully copied URL to clipboard', { timeout: 3000 })
+                    } catch (error) {
+                      reportError('Error copying to clipboard', error)
+                    }
+                  }
+                }, ['Copy URL to clipboard'])
+              ]),
+              side: 'bottom'
+            }, [
+              h(Clickable, { 'aria-label': 'Notebook options' }, [icon('ellipsis-v', {})])
```

Comment:
Currently you have to click directly on the icon to open the menu. We should expand the clickable area by having the `PopupTrigger` wrap the `ButtonSecondary`, and remove the unnecessary inner `Clickable`.